File: CleverFerretV2/core/network/src/main/java/com/cleverferret/v2/core/network/resilience/RetryPolicy.java
package com.cleverferret.v2.core.network.resilience;

import com.cleverferret.v2.core.common.error.AppError;

public interface RetryPolicy {
    boolean shouldRetry(AppError error, int attempt);

    long delayMillis(int attempt);

    int maxAttempts();
}

File: CleverFerretV2/core/network/src/main/java/com/cleverferret/v2/core/network/resilience/RetryPolicy.kt
package com.cleverferret.v2.core.network.resilience

import com.cleverferret.v2.core.common.error.AppError

interface RetryPolicy {
    fun shouldRetry(error: AppError, attempt: Int): Boolean
    fun delayMillis(attempt: Int): Long
    fun maxAttempts(): Int
}
